src/test/java/pl/project13/maven/git/GitIntegrationTest.java

```diff
@@ -18,32 +18,68 @@
 package pl.project13.maven.git;
 
 import com.google.common.base.Optional;
+import org.apache.commons.io.FileUtils;
 import org.apache.maven.project.MavenProject;
 import org.eclipse.jgit.api.Git;
 import org.jetbrains.annotations.NotNull;
+import org.junit.After;
 import org.junit.Before;
 
 import java.io.File;
 import java.io.IOException;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Random;
 
 import static org.mockito.internal.util.reflection.Whitebox.setInternalState;
 
 public abstract class GitIntegrationTest {
 
-  final String SANDBOX_DIR = "target/sandbox";
+  /**
+   * Random number generator for generating random sandbox folder names.
+   */
+  private final static Random r = new Random();
+
+  private final static String SANDBOX_DIR = "target" + File.separator + "sandbox" + File.separator;
+
+  /**
+   * Sandbox directory with unique name for current test.
+   */
+  private String currSandbox;
 
   protected GitCommitIdMojo mojo;
   protected FileSystemMavenSandbox mavenSandbox;
 
   @Before
   public void setUp() throws Exception {
-    mavenSandbox = new FileSystemMavenSandbox(SANDBOX_DIR);
+    // generate unique sandbox for this test
+    File sandbox;
+    do {
+      currSandbox = SANDBOX_DIR + "sandbox" + Integer.toString(r.nextInt(Integer.MAX_VALUE));
```

Comment:
could be even better via using ThreadLocalRandom.current() if we want to enable parallel tests